typescript/tslint.json

```diff
@@ -0,0 +1,43 @@
+ {
+    "extends": ["tslint:recommended", "tslint-immutable", "typestrict"],
+    "jsRules": {
+        "quotemark": false
+    },
+    "rules": {
+        "interface-name": {
+            "severity": "error",
+            "options": "never-prefix"
+        },
+        "max-classes-per-file": false,
+        "max-line-length": false,
```

Comment:
Shouldn't we go for a `warning` here to be consistent with our JS style?

Reply:
Assuming we will use both eslint _and_ tslint for TypeScript code, the logic should be we use eslint for everything it can handle, and tslint for everything else, for example rules with advanced type checking or rules where ordinary eslint rules would produce false positives due to not knowing everything the compiler does.

This serves to reduce duplicate error reports between the tools.

One shared eslint config for TypeScript and JavaScript will become a lot easier with eslint 6 (currently in alpha) and a future compatible release of <https://github.com/typescript-eslint/typescript-eslint>. pickware-cloud does this the hard way, currently.

See <https://www.notion.so/TypeScript-ee03073f3b2345d785a4f90ff35f24f1> for more.

I've discussed this with @fixpunkt in the past.